p2p/net/session.go

```diff
@@ -118,6 +122,20 @@ func (n *NetworkSessionImpl) EncryptGuard() *sync.Mutex {
 	return &n.encGuard
 }
 
+// Sign signs that input message using the sessions hmac and returns the signature or an error
+func (n *NetworkSessionImpl) Sign(data []byte) ([]byte, error) {
```

Comment:
Why create a new hmac form the same key and message each time ? the approach I know is keeping 2 hmacs (in and out) and writing each message then sum again.

Reply:
though we would probably need to lock like we do in the session.

Reply:
I know the approach I mentioned works. https://play.golang.org/p/OtolvAo1gpz

Reply:
This is one of the questions I have for Iddo, I'll address this comment after my talk with him

Reply:
Anyway there are different results for each approach - https://play.golang.org/p/vKYCyAER6qz